packages/oca/src/legacy/resolver/oca.ts

```diff
@@ -188,10 +188,30 @@ export class DefaultOCABundleResolver implements OCABundleResolverType {
   protected bundles: Record<string, OverlayBundle> = {}
   protected options: OCABundleResolverOptions
 
-  public constructor(bundlesData: Record<string, IOverlayBundleData> = {}, options?: OCABundleResolverOptions) {
+  public constructor(
+    bundlesData: Record<string, IOverlayBundleData | string> = {},
+    options?: OCABundleResolverOptions
+  ) {
+    // getting the bundles
     for (const cid in bundlesData) {
       try {
-        this.bundles[cid] = new OverlayBundle(cid, bundlesData[cid])
+        if (typeof bundlesData[cid] !== 'string') {
+          this.bundles[cid] = new OverlayBundle(cid, bundlesData[cid] as IOverlayBundleData)
+        }
+      } catch (error) {
+        // might get an error trying to parse javascript's default value
+      }
+    }
+
+    // getting the bundles from its key
+    for (const cid in bundlesData) {
+      try {
+        if (typeof bundlesData[cid] === 'string') {
```

Comment:
Is there any chance we would encounter have some sort of recursive reference (i.e a bundle id is a reference to another reference)?

Reply:
It might. I'm looking into the code, and there is another place where it checks for aliases (the `resolve` method from the legacy code). The legacy code didn't check for recursions also. It may lead to errors in casting the types in that case. Will do some refactoring to the code to avoid duplication.